RemoveCopyFopyInfo in hpcgap/src/gvars.c is nonsense

GCC produces the following warnings (among others):
```
../../hpcgap/src/gvars.c: In function ‘RemoveCopyFopyInfo’:
../../hpcgap/src/gvars.c:1203:5: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
     l = LEN_PLIST(CopiesGVars);
     ^
../../hpcgap/src/gvars.c:1206:5: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
     l = LEN_PLIST(FopiesGVars);
     ^
```
Upon inspecting the code in question, I think it is bogus:
```
void RemoveCopyFopyInfo( void )
{
    UInt        i, l;

    LockGVars(1);
    l = LEN_PLIST(CopiesGVars);
    for ( i = 1; i <= l; i++ )
        SET_ELM_PLIST( CopiesGVars[GVAR_BUCKET(i)], GVAR_INDEX(i), 0 );
    l = LEN_PLIST(FopiesGVars);
    for ( i = 1; i <= l; i++ )
        SET_ELM_PLIST( FopiesGVars[GVAR_BUCKET(i)], GVAR_INDEX(i), 0 );
    NCopyAndFopyDone = 0;
    UnlockGVars();
    return;
}
```
Note that in HPC-GAP, `CopiesGVars` and `FopiesGVars` are *arrays* of `Obj` objects, so `LEN_PLIST(CopiesGVars)` (the code the compiler warns about) indeed makes no sense. Moreover, I think `CopiesGVars[GVAR_BUCKET(i)]` really makes no sense either.

I have not yet looked closer at how the rewritten gvar code in HPC-GAP works, which is why I am not trying to fix this. My guess is that the two loops each need to be replaced by something like this:
```
  for (k = 0; k < ARRAY_SIZE(CopiesGVars); ++k) {
    l = LEN_PLIST(CopiesGVars[k]);
    for ( i = 1; i <= l; i++ )
        SET_ELM_PLIST( CopiesGVars[k], i, 0 );
  }
```
Then again, this might still be wrong or even nonsense; in any case, something needs to be changed there.

--- a/hpcgap/src/gvars.c
+++ b/hpcgap/src/gvars.c
@@ -59,6 +59,8 @@
 
 #include <src/hpc/systhread.h>          /* system thread primitives */
 
+#include <src/util.h>
+
 #include <stdio.h>
 
 
@@ -1197,15 +1199,26 @@ void UpdateCopyFopyInfo ( void )
 */
 void RemoveCopyFopyInfo( void )
 {
-    UInt        i, l;
+    UInt        i, k, l;
 
     LockGVars(1);
-    l = LEN_PLIST(CopiesGVars);
-    for ( i = 1; i <= l; i++ )
-        SET_ELM_PLIST( CopiesGVars[GVAR_BUCKET(i)], GVAR_INDEX(i), 0 );
-    l = LEN_PLIST(FopiesGVars);
-    for ( i = 1; i <= l; i++ )
-        SET_ELM_PLIST( FopiesGVars[GVAR_BUCKET(i)], GVAR_INDEX(i), 0 );
+
+    for (k = 0; k < ARRAY_SIZE(CopiesGVars); ++k) {
+        if (CopiesGVars[k] == 0)
+            continue;
+        l = LEN_PLIST(CopiesGVars[k]);
+        for ( i = 1; i <= l; i++ )
+            SET_ELM_PLIST( CopiesGVars[k], i, 0 );
+    }
+
+    for (k = 0; k < ARRAY_SIZE(FopiesGVars); ++k) {
+        if (FopiesGVars[k] == 0)
+            continue;
+        l = LEN_PLIST(FopiesGVars[k]);
+        for ( i = 1; i <= l; i++ )
+            SET_ELM_PLIST( FopiesGVars[k], i, 0 );
+    }
+
     NCopyAndFopyDone = 0;
     UnlockGVars();
     return;

